src/Native/Unix/System.Native/pal_process.c

```diff
@@ -386,71 +385,80 @@ static uint64_t ConvertFromNativeRLimitInfinityToManagedIfNecessary(rlim_t value
     if (value == RLIM_INFINITY)
         return UINT64_MAX;
 
-    return UnsignedCast(value);
+    assert(value >= 0);
+    return (uint64_t)value;
 }
 
-static void ConvertFromRLimitManagedToPal(const RLimit& pal, rlimit& native)
+static void ConvertFromRLimitManagedToPal(const RLimit* pal, struct rlimit* native)
 {
-    native.rlim_cur = ConvertFromManagedRLimitInfinityToPalIfNecessary(pal.CurrentLimit);
-    native.rlim_max = ConvertFromManagedRLimitInfinityToPalIfNecessary(pal.MaximumLimit);
+    native->rlim_cur = ConvertFromManagedRLimitInfinityToPalIfNecessary(pal->CurrentLimit);
+    native->rlim_max = ConvertFromManagedRLimitInfinityToPalIfNecessary(pal->MaximumLimit);
 }
 
-static void ConvertFromPalRLimitToManaged(const rlimit& native, RLimit& pal)
+static void ConvertFromPalRLimitToManaged(const struct rlimit* native, RLimit* pal)
 {
-    pal.CurrentLimit = ConvertFromNativeRLimitInfinityToManagedIfNecessary(native.rlim_cur);
-    pal.MaximumLimit = ConvertFromNativeRLimitInfinityToManagedIfNecessary(native.rlim_max);
+    pal->CurrentLimit = ConvertFromNativeRLimitInfinityToManagedIfNecessary(native->rlim_cur);
+    pal->MaximumLimit = ConvertFromNativeRLimitInfinityToManagedIfNecessary(native->rlim_max);
 }
 
-extern "C" int32_t SystemNative_GetRLimit(RLimitResources resourceType, RLimit* limits)
+#if defined __USE_GNU && !defined __cplusplus
+typedef __rlimit_resource_t rlimitResource;
+typedef __priority_which_t priorityWhich;
+#else
+typedef int rlimitResource;
+typedef int priorityWhich;
+#endif
+
+int32_t SystemNative_GetRLimit(RLimitResources resourceType, RLimit* limits)
 {
-    assert(limits != nullptr);
+    assert(limits != NULL);
 
     int32_t platformLimit = ConvertRLimitResourcesPalToPlatform(resourceType);
-    rlimit internalLimit;
-    int result = getrlimit(platformLimit, &internalLimit);
+    struct rlimit internalLimit;
+    int result = getrlimit((rlimitResource)platformLimit, &internalLimit);
     if (result == 0)
     {
-        ConvertFromPalRLimitToManaged(internalLimit, *limits);
+        ConvertFromPalRLimitToManaged(&internalLimit, limits);
     }
     else
     {
-        *limits = {};
+        *limits = (RLimit){0,0};
```

Comment:
Why is the cast here needed?

Reply:
Without this cast, clang was erroring out:

```
/corefx/src/Native/Unix/System.Native/pal_process.c:425:19: error: expected expression
        *limits = {0,0};
                  ^
/corefx/src/Native/Unix/System.Native/pal_process.c:425:19: error: expected expression
        *limits = {0,0};
```

Reply:
Changed to memset in the followup PR: https://github.com/dotnet/corefx/pull/30495.